task2/src/main/java/backend/Teacher.java

```diff
@@ -118,27 +120,52 @@ private double[][] calculateErrorInHiddenLayers() throws IllegalAccessException
         if (neuralNetwork.isWithBias()) {
             biasErrorHidden = new double[numOfHiddenLayers][maxConnections];
         }
+        int numOfConnectionsInAllPreviousLayers = 0;
         for (int hLayer = 0; hLayer < numOfHiddenLayers; hLayer++) {
             int hiddenLayerNum = numOfHiddenLayers - hLayer;
             double[] outputs = neuralNetwork.getLayersResult()[hiddenLayerNum];
             int numOfNeuronsInPreviousLayer = neuralNetwork.getLayersResult()[hiddenLayerNum - 1].length;
+
+            int numOfNeuronsInNextLayer = 0;
+            if (hLayer == 0) {
+                numOfConnectionsInAllPreviousLayers = numOfOutputs;
+                errorDerivative[1] = new double[numOfConnectionsInAllPreviousLayers * outputs.length];
+            } else {
+                numOfNeuronsInNextLayer = neuralNetwork.getLayersResult()[hiddenLayerNum + 1].length;
+                errorDerivative[hLayer + 1] = new double[numOfConnectionsInAllPreviousLayers * outputs.length];
+            }
```

Comment:
This part can be after if 
```suggestion
            if (hLayer == 0) {
                numOfConnectionsInAllPreviousLayers = numOfOutputs;
            } else {
                numOfNeuronsInNextLayer = neuralNetwork.getLayersResult()[hiddenLayerNum + 1].length;
            }
            errorDerivative[hLayer + 1] = new double[numOfConnectionsInAllPreviousLayers * outputs.length];
```